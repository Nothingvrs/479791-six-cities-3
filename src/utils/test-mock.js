export const mockTestData = {
  numberOfOffers: 32,
  cards: [
    `Canal View Prinsengracht`,
    `Nice, cozy, warm big bed apartment`,
    `Wood and stone place`
  ]
};

export const mockCards = [
  {
    id: 0,
    name: `Wood and stone place`,
    type: `Private room`,
    imgs: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/room.jpg`],
    price: 35,
    isInBookmark: false,
    mark: 2,
    isPremium: false,
    bedroomNo: 1,
    capacity: 1,
    facilities: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,

    ],
    descriptions: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ],
    reviews: 5,
    avgMark: 5,
    hostUser: {
      name: `Angelina`,
      img: `img/avatar-angelina.jpg`,
      status: `pro`
    },
    addressCoords: [52.3909553943508, 4.929309666406198]
  },
  {
    id: 1,
    name: `Beautiful & luxurious apartment at great location `,
    type: `Private room`,
    imgs: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/room.jpg`],
    price: 555,
    isInBookmark: true,
    mark: 3,
    isPremium: false,
    bedroomNo: 1,
    capacity: 1,
    facilities: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`
    ],
    descriptions: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ],
    reviews: 2,
    avgMark: 4.5,
    hostUser: {
      name: `Angelina`,
      img: `img/avatar-angelina.jpg`,
      status: `pro`
    },
    addressCoords: [52.369553943508, 4.85309666406198]
  },
  {
    id: 2,
    name: `Canal View Prinsengracht`,
    type: `Apartment`,
    imgs: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/room.jpg`],
    price: 132,
    isInBookmark: false,
    mark: 4,
    isPremium: false,
    bedroomNo: 1,
    capacity: 1,
    facilities: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`
    ],
    descriptions: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ],
    reviews: 2,
    avgMark: 4.5,
    hostUser: {
      name: `Angelina`,
      img: `img/avatar-angelina.jpg`,
      status: `pro`
    },
    addressCoords: [52.3909553943508, 4.929309666406198]
  },
  {
    id: 3,
    name: `Nice, cozy, warm big bed apartment`,
    type: `Apartment`,
    imgs: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/room.jpg`],
    price: 180,
    isInBookmark: false,
    mark: 5,
    isPremium: false,
    bedroomNo: 1,
    capacity: 1,
    facilities: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`
    ],
    descriptions: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ],
    reviews: 2,
    avgMark: 4.5,
    hostUser: {
      name: `Angelina`,
      img: `img/avatar-angelina.jpg`,
      status: `pro`
    },
    addressCoords: [52.3809553943508, 4.939309666406198]
  }
];


export const findByTestAtr = (component, attr) => {
  return component.find(`[data-test='${attr}']`);
};


