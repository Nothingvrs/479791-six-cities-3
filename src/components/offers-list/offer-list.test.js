import renderer from 'react-test-renderer';
import React from 'react';
import OffersList from './offers-list';
import {mockCards} from '../../utils/test-mock.js';

it(`OffersList successfully rendered`, () => {
  const tree = renderer.create(<OffersList cards={mockCards} onCardHover={() => {}} />);
  expect(tree).toMatchSnapshot();
});
