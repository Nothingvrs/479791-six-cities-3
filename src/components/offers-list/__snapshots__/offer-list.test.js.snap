// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card successfully successfully rendered 1`] = `
Array [
  <Connect(withRouter(OfferCard))
    card={
      Object {
        "addressCoords": Array [
          52.3909553943508,
          4.85309666406198,
        ],
        "avgMark": 4.5,
        "bedroomNo": 1,
        "capacity": 1,
        "city": Object {
          "location": Object {
            "latitude": 52.37454,
            "longitude": 4.897976,
            "zoom": 13,
          },
          "name": "Amsterdam",
        },
        "comments": Array [
          Object {
            "author": "Max",
            "authorImg": "/img/avatar-max.jpg",
            "comment": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
            "date": "2020-01-26T13:51:50.417Z",
            "id": 1,
            "mark": 4,
          },
          Object {
            "author": "Max",
            "authorImg": "/img/avatar-max.jpg",
            "comment": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
            "date": "2020-01-26T13:51:50.417Z",
            "id": 2,
            "mark": 4,
          },
        ],
        "descriptions": Array [
          "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
          "An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.",
        ],
        "facilities": Array [
          "Wi-Fi",
          "Heating",
          "Kitchen",
          "Fridge",
          "Washing machine",
          "Coffee machine",
          "Dishwasher",
        ],
        "hostUser": Object {
          "img": "/img/avatar-angelina.jpg",
          "isPro": true,
          "name": "Angelina",
        },
        "id": 0,
        "imgs": Array [
          "/img/apartment-01.jpg",
          "/img/apartment-02.jpg",
          "/img/apartment-03.jpg",
          "/img/studio-01.jpg",
          "/img/room.jpg",
        ],
        "isInBookmark": true,
        "isPremium": true,
        "mark": 4,
        "name": "Beautiful & luxurious apartment at great location",
        "nearOffers": Array [
          1,
          2,
          3,
        ],
        "previewImg": "/img/apartment-01.jpg",
        "price": 120,
        "reviews": 2,
        "type": "Apartment",
      }
    }
    hovered={false}
    onHover={[Function]}
    onUnHover={[Function]}
  >
    <article
      className="cities__place-card  place-card"
      data-test="test-card-hover"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="place-card__mark"
      >
        <span>
          Premium
        </span>
      </div>
      <div
        className="cities__image-wrapper  place-card__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="200"
            src="/img/apartment-01.jpg"
            width="260"
          />
        </a>
      </div>
      <div
        className=" place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              120
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <Link
            onClick={[Function]}
            to="/login"
          >
            <LinkAnchor
              href="/login"
              navigate={[Function]}
              onClick={[Function]}
            >
              <a
                href="/login"
                onClick={[Function]}
              >
                <button
                  className="place-card__bookmark-button button place-card__bookmark-button--active"
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    In
                     bookmarks
                  </span>
                </button>
              </a>
            </LinkAnchor>
          </Link>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "90%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <Link
            data-test="test-header-click"
            href="#"
            to="/offer/0"
          >
            <LinkAnchor
              data-test="test-header-click"
              href="/offer/0"
              navigate={[Function]}
            >
              <a
                data-test="test-header-click"
                href="/offer/0"
                onClick={[Function]}
              >
                Beautiful & luxurious apartment at great location
              </a>
            </LinkAnchor>
          </Link>
        </h2>
        <p
          className="place-card__type"
        >
          Apartment
        </p>
      </div>
    </article>
  </Connect(withRouter(OfferCard))>,
  <Connect(withRouter(OfferCard))
    card={
      Object {
        "addressCoords": Array [
          52.369553943508,
          4.85309666406198,
        ],
        "avgMark": 4.5,
        "bedroomNo": 3,
        "capacity": 4,
        "city": Object {
          "location": Object {
            "latitude": 52.37454,
            "longitude": 4.897976,
            "zoom": 13,
          },
          "name": "Amsterdam",
        },
        "comments": Array [
          Object {
            "author": "Max",
            "authorImg": "/img/avatar-max.jpg",
            "comment": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
            "date": "2020-01-26T13:51:50.417Z",
            "id": 2,
            "mark": 4,
          },
        ],
        "descriptions": Array [
          "An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.",
        ],
        "facilities": Array [
          "Wi-Fi",
          "Heating",
          "Kitchen",
          "Fridge",
          "Washing machine",
          "Coffee machine",
          "Dishwasher",
          "Towels",
          "Baby seat",
        ],
        "hostUser": Object {
          "img": "/img/avatar-angelina.jpg",
          "isPro": true,
          "name": "Natalia",
        },
        "id": 1,
        "imgs": Array [
          "/img/room.jpg",
          "/img/apartment-01.jpg",
          "/img/apartment-02.jpg",
          "/img/apartment-03.jpg",
          "/img/studio-01.jpg",
          "/img/room.jpg",
        ],
        "isInBookmark": true,
        "isPremium": false,
        "mark": 0,
        "name": "Wood and stone place",
        "nearOffers": Array [
          0,
          2,
          3,
        ],
        "previewImg": "/img/apartment-01.jpg",
        "price": 80,
        "reviews": 2,
        "type": "Private room",
      }
    }
    hovered={false}
    onHover={[Function]}
    onUnHover={[Function]}
  >
    <article
      className="cities__place-card  place-card"
      data-test="test-card-hover"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="cities__image-wrapper  place-card__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="200"
            src="/img/apartment-01.jpg"
            width="260"
          />
        </a>
      </div>
      <div
        className=" place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              80
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <Link
            onClick={[Function]}
            to="/login"
          >
            <LinkAnchor
              href="/login"
              navigate={[Function]}
              onClick={[Function]}
            >
              <a
                href="/login"
                onClick={[Function]}
              >
                <button
                  className="place-card__bookmark-button button place-card__bookmark-button--active"
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    In
                     bookmarks
                  </span>
                </button>
              </a>
            </LinkAnchor>
          </Link>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "90%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <Link
            data-test="test-header-click"
            href="#"
            to="/offer/1"
          >
            <LinkAnchor
              data-test="test-header-click"
              href="/offer/1"
              navigate={[Function]}
            >
              <a
                data-test="test-header-click"
                href="/offer/1"
                onClick={[Function]}
              >
                Wood and stone place
              </a>
            </LinkAnchor>
          </Link>
        </h2>
        <p
          className="place-card__type"
        >
          Private room
        </p>
      </div>
    </article>
  </Connect(withRouter(OfferCard))>,
  <Connect(withRouter(OfferCard))
    card={
      Object {
        "addressCoords": Array [
          52.3909553943508,
          4.929309666406198,
        ],
        "avgMark": 4.5,
        "bedroomNo": 3,
        "capacity": 4,
        "city": Object {
          "location": Object {
            "latitude": 52.37454,
            "longitude": 4.897976,
            "zoom": 13,
          },
          "name": "Amsterdam",
        },
        "comments": Array [
          Object {
            "author": "Max",
            "authorImg": "/img/avatar-max.jpg",
            "comment": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
            "date": "2020-01-26T13:51:50.417Z",
            "id": 1,
            "mark": 4,
          },
          Object {
            "author": "Max",
            "authorImg": "/img/avatar-max.jpg",
            "comment": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
            "date": "2020-01-26T13:51:50.417Z",
            "id": 2,
            "mark": 4,
          },
        ],
        "descriptions": Array [
          "Chalets Alpins- 15 Chemin des Skieurs is located in Stoneham, 9.7 km from Les Marais du Nord, 9.7 km from Camping Stoneham, as well as 19.3 km from Nordique Spa Stoneham. The air-conditioned accommodation is an 8-minute walk from Halte O Spa, and guests benefit from private parking available on site and free WiFi.",
          "Aventures Nord-Bec is 24.1 km from the chalet, while Jacques-Cartier National Park is 48.3 km from the property. The nearest airport is Québec City Jean Lesage International Airport, 40.2 km from Chalets Alpins- 15.",
        ],
        "facilities": Array [
          "Wi-Fi",
          "Heating",
          "Kitchen",
          "Fridge",
          "Washing machine",
          "Coffee machine",
          "Dishwasher",
          "Towels",
          "Baby seat",
          "Cabel TV",
        ],
        "hostUser": Object {
          "img": "/img/avatar-angelina.jpg",
          "isPro": true,
          "name": "Olga",
        },
        "id": 2,
        "imgs": Array [
          "/img/apartment-02.jpg",
          "/img/apartment-01.jpg",
          "/img/apartment-02.jpg",
          "/img/apartment-03.jpg",
          "/img/studio-01.jpg",
          "/img/room.jpg",
        ],
        "isInBookmark": false,
        "isPremium": false,
        "mark": 4,
        "name": "Canal View Prinsengracht",
        "nearOffers": Array [
          0,
          1,
          3,
        ],
        "previewImg": "/img/apartment-01.jpg",
        "price": 132,
        "reviews": 2,
        "type": "Apartment",
      }
    }
    hovered={false}
    onHover={[Function]}
    onUnHover={[Function]}
  >
    <article
      className="cities__place-card  place-card"
      data-test="test-card-hover"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="cities__image-wrapper  place-card__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="200"
            src="/img/apartment-01.jpg"
            width="260"
          />
        </a>
      </div>
      <div
        className=" place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              132
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <Link
            onClick={[Function]}
            to="/login"
          >
            <LinkAnchor
              href="/login"
              navigate={[Function]}
              onClick={[Function]}
            >
              <a
                href="/login"
                onClick={[Function]}
              >
                <button
                  className="place-card__bookmark-button button false"
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To
                     bookmarks
                  </span>
                </button>
              </a>
            </LinkAnchor>
          </Link>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "90%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <Link
            data-test="test-header-click"
            href="#"
            to="/offer/2"
          >
            <LinkAnchor
              data-test="test-header-click"
              href="/offer/2"
              navigate={[Function]}
            >
              <a
                data-test="test-header-click"
                href="/offer/2"
                onClick={[Function]}
              >
                Canal View Prinsengracht
              </a>
            </LinkAnchor>
          </Link>
        </h2>
        <p
          className="place-card__type"
        >
          Apartment
        </p>
      </div>
    </article>
  </Connect(withRouter(OfferCard))>,
  <Connect(withRouter(OfferCard))
    card={
      Object {
        "addressCoords": Array [
          52.3809553943508,
          4.939309666406198,
        ],
        "avgMark": 4.5,
        "bedroomNo": 3,
        "capacity": 4,
        "city": Object {
          "location": Object {
            "latitude": 52.37454,
            "longitude": 4.897976,
            "zoom": 13,
          },
          "name": "Amsterdam",
        },
        "comments": Array [
          Object {
            "author": "Alex",
            "authorImg": "/img/avatar-max.jpg",
            "comment": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
            "date": "2020-03-26T13:51:50.417Z",
            "id": 1,
            "mark": 1,
          },
          Object {
            "author": "Max",
            "authorImg": "/img/avatar-max.jpg",
            "comment": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
            "date": "2020-01-26T13:51:50.417Z",
            "id": 2,
            "mark": 4,
          },
        ],
        "descriptions": Array [
          "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
          "An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.",
          "The chalet features 5 separate bedrooms, 1 bathroom, a fully equipped kitchen with a dining area and dishwasher, and a living room with a TV.",
        ],
        "facilities": Array [
          "Wi-Fi",
          "Heating",
          "Kitchen",
          "Fridge",
          "Washing machine",
          "Coffee machine",
          "Dishwasher",
          "Towels",
          "Baby seat",
          "Cabel TV",
        ],
        "hostUser": Object {
          "img": "/img/avatar-angelina.jpg",
          "isPro": true,
          "name": "Natalia",
        },
        "id": 3,
        "imgs": Array [
          "/img/apartment-03.jpg",
          "/img/apartment-01.jpg",
          "/img/apartment-02.jpg",
          "/img/apartment-03.jpg",
          "/img/studio-01.jpg",
          "/img/room.jpg",
        ],
        "isInBookmark": false,
        "isPremium": false,
        "mark": 5,
        "name": "Nice, cozy, warm big bed apartment",
        "nearOffers": Array [
          0,
          1,
          2,
        ],
        "previewImg": "/img/apartment-01.jpg",
        "price": 180,
        "reviews": 2,
        "type": "Apartment",
      }
    }
    hovered={false}
    onHover={[Function]}
    onUnHover={[Function]}
  >
    <article
      className="cities__place-card  place-card"
      data-test="test-card-hover"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="cities__image-wrapper  place-card__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="200"
            src="/img/apartment-01.jpg"
            width="260"
          />
        </a>
      </div>
      <div
        className=" place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              180
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <Link
            onClick={[Function]}
            to="/login"
          >
            <LinkAnchor
              href="/login"
              navigate={[Function]}
              onClick={[Function]}
            >
              <a
                href="/login"
                onClick={[Function]}
              >
                <button
                  className="place-card__bookmark-button button false"
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To
                     bookmarks
                  </span>
                </button>
              </a>
            </LinkAnchor>
          </Link>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "90%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <Link
            data-test="test-header-click"
            href="#"
            to="/offer/3"
          >
            <LinkAnchor
              data-test="test-header-click"
              href="/offer/3"
              navigate={[Function]}
            >
              <a
                data-test="test-header-click"
                href="/offer/3"
                onClick={[Function]}
              >
                Nice, cozy, warm big bed apartment
              </a>
            </LinkAnchor>
          </Link>
        </h2>
        <p
          className="place-card__type"
        >
          Apartment
        </p>
      </div>
    </article>
  </Connect(withRouter(OfferCard))>,
]
`;
